content/cucumber/reporting.md

```diff
@@ -235,6 +236,137 @@ Alternatively, you can specify a Java system property (in Maven, Gradle or other
 ```
 {{% /block %}}
 
+# Custom formatters
+The Cucumber formatter API is readily extensible. A formatter can be any class implementing the event-based formatter
+API. {{% text "ruby" %}}The formatter class should live in the `features/support` directory. {{% /text %}}To use your custom formatter, run Cucumber using the `--format` flag:
+```
+cucumber --format CustomFormatter
+```
+
+## Formatter API
+Cucumber uses an event-based API for its formatters. These formatters respond to several defined events, with
+event handlers defined in the formatter's constructor. 
+{{% block "ruby" %}}
+A sample formatter could look like this:
+```ruby
+class CustomFormatter
+  attr_reader :io
+
+  def initialize(config)
+    @io = config.out_stream
+
+    # Using a block
+    config.on_event :test_step_started do |event|
+      io.puts 'Started test step'
+    end
+
+    # Passing in a method
+    config.on_event :test_step_finished &method(:print_finished)
+  end
+
+  def print_finished(event)
+    io.puts 'Finished test step'
+  end
+end
+```
+{{% /block %}}
+
+### Configuration object
+The formatter initializer is passed a {{% text "java,dotnet" %}}Cucumber configuration{{% /text %}}{{% text "ruby" %}}`Cucumber::Configuration`{{% /text %}} object. This is the 
+[configuration for the test run](https://docs.cucumber.io/cucumber/configuration/),
+including default configurations and [options](https://docs.cucumber.io/cucumber/api/) passed in at the command line.
```

Comment:
Looks like "options" is linking to the general /api page; I think it should link to a particular heading. If you don't know how, I can help with that. Could you find the correct heading to link to?